asserts/asserts_test.go

```diff
@@ -0,0 +1,164 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2014-2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package asserts_test
+
+import (
+	"strings"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/ubuntu-core/snappy/asserts"
+)
+
+type AssertsSuite struct{}
+
+var _ = Suite(&AssertsSuite{})
+
+func (as *AssertsSuite) TestDecodeEmptyBody(c *C) {
+	encoded := "type: test-only\n" +
+		"revision: 0\n" +
```

Comment:
I'd standardize having the revision after the primary key, so:

```
type (what is this)
authority-id (who said so)
primary-key (about what)
revision (how recently)
cruft (whatever)
```

It also makes me wonder again about the fact revision may not be always there. Might be worth going over the list of known assertions in the design doc. I'm almost sure we talked about one case where there was no revision. Can't remember the reason now, but looking at the assertion I probably will.